src/frontmatter.md

```diff
@@ -0,0 +1,42 @@
+r[frontmatter]
+# Frontmatter
+
+r[frontmatter.syntax]
+```grammar,lexer
+FRONTMATTER ->
+      FRONTMATTER_FENCE INFOSTRING? LF
+      (FRONTMATTER_LINE LF )*
+      FRONTMATTER_FENCE[^matched-fence] LF
+
+FRONTMATTER_FENCE -> `---` `-`*
+
+INFOSTRING -> XID_Start ( XID_Continue | `-` | `.` )*
```

Comment:
`XID_Start` doesn't include `_`, but the current implementation does allow infostrings to start with `_`.

Reply:
Thanks for catching that!